src/pages/demo.tsx

```diff
@@ -20,22 +20,14 @@ export const DemoLayout: FunctionComponent = ({ children }) => {
 };
 
 export const Demo: FunctionComponent = () => {
-  const { login, isLoggedIn } = useAuthContext();
-  const { upgradeToMagicSigner } = useProviderContext();
-  const history = useHistory();
+  const dispatch = useDispatch();
 
-  useEffect(() => {
-    const handleIsLoggedIn = async () => {
-      if (isLoggedIn) {
-        await upgradeToMagicSigner();
-        history.push("/demo/create");
-      }
-    };
+  React.useLayoutEffect(() => {
+    dispatch(resetDemoCreateState());
+    dispatch(resetDemoVerifyState());
+  }, [dispatch]);
```

Comment:
um dont really understand how useLayoutEffect works but seems working

Reply:
i read the diff is about the timing of when changes is painted on screen, 1 is earlier, 1 is later 🤔